Updating

s4nch3z opened the pull request.

Commit:
The AWS Mobile SDK for iOS 2.4.2.

Commit:
fix warnings on 32 bit devices

Commit:
The AWS Mobile SDK for iOS 2.4.3.

Commit:
The AWS Mobile SDK for iOS 2.4.4.

karthiksaligrama committed:
Update CHANGELOG.md

Commit:
The AWS Mobile SDK for iOS 2.4.5.

Commit:
The AWS Mobile SDK for iOS 2.4.6.

Commit:
The AWS Mobile SDK for iOS 2.4.7.

Commit:
The AWS Mobile SDK for iOS 2.4.8.

karthiksaligrama committed:
Update AWSCore.podspec

arvindravi committed:
Fixed reference to AWSURLRequestRetryHandler

arvindravi committed:
Fixed Reference Syntax

karthiksaligrama committed:
Merge pull request #462 from arvindravi/master

Fixed reference to AWSURLRequestRetryHandler

Commit:
The AWS Mobile SDK for iOS 2.4.9.

Commit:
The AWS Mobile SDK for iOS 2.4.10.

Commit:
The AWS Mobile SDK for iOS 2.4.11.

Commit:
The AWS Mobile SDK for iOS 2.4.12.

Commit:
The AWS Mobile SDK for iOS 2.4.13.

Commit:
The AWS Mobile SDK for iOS 2.4.14.

Commit:
The AWS Mobile SDK for iOS 2.4.15.

Commit:
The AWS Mobile SDK for iOS 2.4.16.

jamesbeith committed:
Fix parameter not found in function declaration

karthiksaligrama committed:
Merge pull request #554 from jamesbeith/patch-1

Fix parameter not found in function declaration

Commit:
Allow for future expired/unauthed token calls to properly refresh the token

Commit:
The AWS Mobile SDK for iOS 2.5.0.

minbi committed:
Update CHANGELOG.md

minbi committed:
Update CHANGELOG.md

Commit:
The AWS Mobile SDK for iOS 2.5.1.

minbi committed:
Update CHANGELOG.md

nahashsvb committed:
Fixed crash in (nextChunk) method (#582)

If no data is read crash observed at
“[NSData dataWithBytesNoCopy:chunkBuffer length:read];” string.

unsigned int "read" was converted to signed int creating "absurd" length when attempting to create data object.

karthiksaligrama committed:
Update README.md

karthiksaligrama committed:
Update README.md

Commit:
The AWS Mobile SDK for iOS 2.5.2.

ramki982 committed:
Fix issue reported in Shadow Operation Still In Progress! BUG! #565 (#601)

minbi committed:
Update CHANGELOG.md

Remove change that did not go out.

Commit:
Fix space between headers and text

mlshon committed:
Merge pull request #622 from miguelarauj1o/patch-1

Fix space between headers and text

Commit:
The AWS Mobile SDK for iOS 2.5.3.

lyricsboy committed:
Fixes a typo that caused a bug when attempting to access the rawResponse.

The `rawReponse` property would never be non-nil, and the actual `rawResponse` property was inaccessible, particularly in Swift.

harshavardhana committed:
Remove Transfer-Encoding from streaming signature

http://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-streaming.html 
doesn't provide any detail related to `Transfer-Encoding` to be a valid 
header as a requirement. 

Additionally also Transfer-Encoding is used as part of signature 
request incorrectly because the transfer coding names are case 
insensitive[1] i.e when the canonical request is generated should
be generated as lower cased value instead.

```
content-encoding:aws-chunked
content-type:image/jpeg
host:***REDACTED***
transfer-encoding:Chunked
user-agent:aws-sdk-iOS/2.5.2 iOS/10.2 en_US transfer-manager
x-amz-acl:public-read
x-amz-content-sha256:STREAMING-AWS4-HMAC-SHA256-PAYLOAD
x-amz-date:20170413T050829Z
x-amz-decoded-content-length:1024000
```

[1] - http://greenbytes.de/tech/webdav/rfc7230.html#transfer.codings

bryant1410 committed:
Fix broken Markdown headings

mlshon committed:
Merge pull request #639 from bryant1410/master

Fix broken headings in Markdown files

Commit:
The AWS Mobile SDK for iOS 2.5.4.

rohandubal committed:
Update CHANGELOG.md

Commit:
The AWS Mobile SDK for iOS 2.5.4.

minbi committed:
Update CHANGELOG.md

Commit:
The AWS Mobile SDK for iOS 2.5.5.

karthiksaligrama committed:
Logging updates to README.md

Commit:
The AWS Mobile SDK for iOS 2.5.6.

SlaunchaMan committed:
Fix documentation typo.

The parameter names should match.

rohandubal committed:
Merge pull request #563 from donileo/master

Allow for future expired/unauthed token calls to properly refresh the aws token.

Commit:
The AWS Mobile SDK for iOS 2.5.7.

rohandubal committed:
Update CHANGELOG.md

karthiksaligrama committed:
Merge pull request #665 from SlaunchaMan/patch-1

Fix documentation typo.

makarupang reviewed (approved):
for all lost commant

lyricsboy committed:
Merge branch 'master' into lyricsboy/rawResponse-rename

Commit:
The AWS Mobile SDK for iOS 2.5.8.

kishorepran committed:
Fix Documentation typo. Shows as warning in Xcode 8.2.3

Commit:
Update AWSiOSSDKv2.podspec

Commit:
add method overload to allow specifying partition key explicitly

Commit:
The AWS Mobile SDK for iOS 2.5.9.

Commit:
Fixing formatting in CHANGELOG and adding a description of Lex changes

lukman-foreflight committed:
AWSS3PreSignedURLBuilder configuration copy

only create a new copy of configuration.endpoint if one does not exist
already.

joshuafarray committed:
Fixed minor documentation typos

lukman-foreflight committed:
AWSS3PreSignedURLBuilder configuration copy

set AWSServiceS3 as service on configuration for endpoint if it already
exist.

Commit:
Merge pull request #715 from joshuafarray/master

Fixed minor documentation typos

Commit:
Merge pull request #685 from kishorepran/master

Fix Documentation typo. Shows as warning in Xcode 8.2.3

noremac committed:
Fixes compilation errors in AWSDDODLogger reported by Xcode 9.

Commit:
The AWS Mobile SDK for iOS 2.5.10.

minbi committed:
Update CHANGELOG.md

rohandubal committed:
Merge pull request #709 from foreflight/pre_signed_url_builder_endpoint_copy

AWSS3PreSignedURLBuilder configuration copy

rohandubal committed:
Merge pull request #719 from noremac/xcode9

Fixes compilation errors in AWSDDODLogger reported by Xcode 9.

rohandubal committed:
Merge pull request #631 from qblu/lyricsboy/rawResponse-rename

Fixes a typo/bug when accessing AWSAPIGatewayResponse.rawResponse

Commit:
The AWS Mobile SDK for iOS 2.6.0.

rohandubal committed:
Update CHANGELOG.md

rohandubal committed:
Update CHANGELOG.md

Commit:
The AWS Mobile SDK for iOS 2.6.1.

rohandubal committed:
Update CHANGELOG.md

Commit:
The AWS Mobile SDK for iOS 2.6.2.

rohandubal committed:
Update CHANGELOG.md

minbi committed:
Merge branch 'master' into feature/partition-key-recorder

minbi committed:
Merge pull request #696 from igor-makarov/feature/partition-key-recorder

add method overload to allow specifying partition key explicitly

minbi committed:
Merge pull request #638 from harshavardhana/patch-1

Remove Transfer-Encoding from streaming signature
